### PR TITLE
Adding support for managed policies

### PR DESCRIPTION
**Pull request by shaunbrady**
Amazon recently added managed policies for IAM resources.  Open issue: https://github.com/boto/boto/issues/2956.  This Pull Request covers each of the listed API calls.

### CODE REVIEW

**Comment by mfulleratlassian** [comment]
Any chance this will get merged?

**Comment by amadornimbis** [comment]
:+1: Seconding the "will this get merged?"

**Comment by kdimiche** [comment]
Requesting merge

**Comment by kyleknap** [comment]
We will need to verify the update matches IAM's API.

**Comment by shaunbrady** [comment]
@kyleknap Thank you very much.  Please see the provided tests (unit and integration).  Let me know if you need any thing else.

**Comment by kyleknap** [comment]
We should be good for now. The verification is a matter of us looking up the methods you added and making sure they match up with the API reference docs.

**Comment by shaunbrady** [comment]
Any word on this?  We're likely moving forward with an AWS usage and managed policies would greatly simplify the code/user management.

**Comment by lightmanca** [comment]
Hey how about a merge?  I need this too.

**Comment by eleniS** [comment]
+1.

File: boto/iam/connection.py [hunk 1]
```diff
@@ -1640,3 +1640,286 @@ def update_account_password_policy(self, allow_users_to_change_password=None,
         if require_uppercase_characters is not None and type(allow_users_to_change_password) is bool:
             params['RequireUppercaseCharacters'] = str(require_uppercase_characters).lower()
         return self.get_response('UpdateAccountPasswordPolicy', params)
+
+    def create_policy(self, policy_name, policy_document, path='/'):
```
Comment (garnaat) [inline boto/iam/connection.py]: This should have an optional description param.

File: tests/unit/iam/test_policy.py [hunk 2]
```diff
@@ -0,0 +1,680 @@
...
+      <Arn>arn:aws:iam::123456789012:policy/S3-read-only-example-bucket</Arn>
+      <AttachmentCount>9</AttachmentCount>
+      <CreateDate>2014-09-15T17:36:14Z</CreateDate>
+      <UpdateDate>2014-09-15T20:31:47Z</UpdateDate>
+    </Policy>
+  </GetPolicyResult>
+  <ResponseMetadata>
+    <RequestId>684f0917-3d22-11e4-a4a0-cffb9EXAMPLE</RequestId>
+  </ResponseMetadata>
+</GetPolicyResponse>
```
Comment (garnaat) [inline tests/unit/iam/test_policy.py]: This response could contain a ``<Description>`` element.  Might want to include it here or test both cases.

**Comment by garnaat** [comment]
I commented on one small discrepancy.  Otherwise, it looks good to me.

**Comment by shaunbrady** [comment]
Good catch.  I have the method and the test updated.  Let me know if you need any thing else.  Thanks.

**Comment by jonathanwcrane** [comment]
Looking forward to this eagerly.

**Comment by kyleknap** [comment]
Thanks for all of the work here! Merging.

**Comment by jonathanwcrane** [comment]
Oh hell yes!

**Comment by webratz** [comment]
this didn't make it yet to the current release on pypi, right?

**Comment by phspagiari** [comment]
This PR and Boto3 doesn't support to update an managed policy, right?

**Pull request merged**